lib/web/databases.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package web
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/gravitational/trace"
+	"github.com/julienschmidt/httprouter"
+
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/reversetunnel"
+	dbiam "github.com/gravitational/teleport/lib/srv/db/common/iam"
+)
+
+// databaseIAMPolicyResponse is the response type for handleDatabaseGetIAMPolicy.
+type databaseIAMPolicyResponse struct {
+	// Type is the type of the IAM policy.
+	Type string `json:"type"`
+	// AWS contains the IAM policy for AWS-hosted databases.
+	AWS *databaseIAMPolicyAWS `json:"aws,omitempty"`
+}
+
+// databaseIAMPolicyAWS contains IAM policy for AWS-hosted databases.
+type databaseIAMPolicyAWS struct {
+	// PolicyDocument is the AWS IAM policy document.
+	PolicyDocument string `json:"policy_document"`
+	// Placeholders are placeholders found in the policy document.
+	Placeholders []string `json:"placeholders,omitempty"`
+}
+
+// handleDatabaseGetIAMPolicy returns the required IAM policy for database.
+func (h *Handler) handleDatabaseGetIAMPolicy(w http.ResponseWriter, r *http.Request, p httprouter.Params, ctx *SessionContext, site reversetunnel.RemoteSite) (interface{}, error) {
+	databaseName := p.ByName("database")
+	if databaseName == "" {
+		return nil, trace.BadParameter("missing database name")
+	}
+
+	clt, err := ctx.GetUserClient(site)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	database, err := fetchDatabaseWithName(r.Context(), clt, r, databaseName)
```

Comment:
I wanted to fetch database servers because:

- to validate the database is successfully added
- there were plans to add aws identity to the heartbeat so we can return the aws identity here

But if you have a plan to add another API for getting database servers, i can do `GetDatabase` here.

Reply:
Sorry, I was not clear and just found out I was suggesting a non-existent method :facepalm: 

So, I was trying to ask why are we using the ListResources method instead of calling the `GetDatabaseServer`.
Turns out, there's no `GetDatabaseServer` but only the _list_ method: `GetDatabaseServers`

On another note, I do plan on having a `GET .../database_servers/:db_server_name` later on
Do you think this AWS info is better suited for a `.../database_servers/:db` or a `.../databases/:db`

Reply:
I think types.Database is sufficient if
- types.DatabaseServer will be validated through another API
- this API is only intended for the discover flow (since types.Database is only for web/tctl added databases)
I will switch.